### PR TITLE
chore: Fix linter findings for `revive:enforce-slice-style` in `plugins/inputs/[p-z]*`

### PR DESCRIPTION
**Pull request by zak-pawel**
## Summary

Address findings for [revive:enforce-slice-style](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#enforce-slice-style) in `plugins/inputs/[p-z]*`.

As we agreed here https://github.com/influxdata/telegraf/issues/15812, we will fix the errors found by the following configuration:
```
      - name: enforce-slice-style
        arguments: ["make"]
```

It is only part of the bigger work.
After all findings of this type in whole project are handled, we can enable `revive:enforce-slice-style` rule in `golangci-lint`.

## Checklist

- [x] No AI generated code was used in this PR

### CODE REVIEW

**Review by DStrand1 — APPROVED** [review]

File: plugins/inputs/smart/smart_test.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func TestGatherInParallelMode(t *testing.T) {
 		require.NoError(t, err)
 
 		result := acc.GetTelegrafMetrics()
-		testutil.RequireMetricsEqual(t, []telegraf.Metric{}, result)
+		testutil.RequireMetricsEqual(t, make([]telegraf.Metric, 0), result)
```
Comment (srebhan) [inline plugins/inputs/smart/smart_test.go:143]:
```suggestion
		require.Empty(t, result)
```

File: plugins/inputs/snmp/snmp_test.go [hunk 2]
```diff
@@ -676,7 +676,7 @@ func TestSnmpInit_noTranslateGosmi(t *testing.T) {
 				}},
 		},
 		ClientConfig: snmp.ClientConfig{
-			Path:       []string{},
+			Path:       make([]string, 0),
```
Comment (srebhan) [inline plugins/inputs/snmp/snmp_test.go:679]: Remove?

File: plugins/inputs/sqlserver/sqlserver_test.go [hunk 3]
```diff
@@ -16,7 +16,7 @@ import (
 func TestSqlServer_QueriesInclusionExclusion(t *testing.T) {
 	cases := []map[string]interface{}{
 		{
-			"IncludeQuery": []string{},
+			"IncludeQuery": make([]string, 0),
```
Comment (srebhan) [inline plugins/inputs/sqlserver/sqlserver_test.go:19]: Remove?

File: plugins/inputs/stackdriver/stackdriver.go [hunk 4]
```diff
@@ -718,7 +718,7 @@ func init() {
 			RateLimit:                       defaultRateLimit,
 			Delay:                           defaultDelay,
 			GatherRawDistributionBuckets:    true,
-			DistributionAggregationAligners: []string{},
+			DistributionAggregationAligners: make([]string, 0),
```
Comment (srebhan) [inline plugins/inputs/stackdriver/stackdriver.go:721]: Remove?

File: plugins/inputs/supervisor/supervisor_test.go [hunk 5]
```diff
@@ -104,8 +104,8 @@ func TestShort_SampleData(t *testing.T) {
 		t.Run(tC.desc, func(t *testing.T) {
 			s := &Supervisor{
 				Server:     "http://example.org:9001/RPC2",
-				MetricsInc: []string{},
-				MetricsExc: []string{},
+				MetricsInc: make([]string, 0),
+				MetricsExc: make([]string, 0),
```
Comment (srebhan) [inline plugins/inputs/supervisor/supervisor_test.go:108]: Remove?

File: plugins/inputs/supervisor/supervisor_test.go [hunk 6]
```diff
@@ -153,8 +153,8 @@ func TestIntegration_BasicGathering(t *testing.T) {
 	defer ctr.Terminate()
 	s := &Supervisor{
 		Server:     "http://login:pass@" + testutil.GetLocalHost() + ":" + ctr.Ports[supervisorPort] + "/RPC2",
-		MetricsInc: []string{},
-		MetricsExc: []string{},
+		MetricsInc: make([]string, 0),
+		MetricsExc: make([]string, 0),
```
Comment (srebhan) [inline plugins/inputs/supervisor/supervisor_test.go:157]: Remove?

File: plugins/inputs/varnish/varnish.go [hunk 7]
```diff
@@ -408,7 +408,7 @@ func init() {
 			UseSudo:         false,
 			InstanceName:    "",
 			Timeout:         defaultTimeout,
-			Regexps:         []string{},
+			Regexps:         make([]string, 0),
```
Comment (srebhan) [inline plugins/inputs/varnish/varnish.go:411]: Remove?

File: plugins/inputs/vsphere/vsan.go [hunk 8]
```diff
@@ -388,7 +388,7 @@ func (e *Endpoint) queryResyncSummary(ctx context.Context, vsanClient *soap.Clie
 	includeSummary := true
 	request := vsantypes.VsanQuerySyncingVsanObjects{
 		This:           vsanSystemEx,
-		Uuids:          []string{}, // We only need summary information.
+		Uuids:          make([]string, 0), // We only need summary information.
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:391]: Remove?

File: plugins/inputs/zipkin/handler_test.go [hunk 9]
```diff
@@ -63,7 +63,7 @@ func TestSpanHandler(t *testing.T) {
 			Timestamp:   time.Unix(0, 1498688360851331*int64(time.Microsecond)).UTC(),
 			Duration:    time.Duration(53106) * time.Microsecond,
 			ServiceName: "trivial",
-			Annotations: []trace.Annotation{},
+			Annotations: make([]trace.Annotation, 0),
```
Comment (srebhan) [inline plugins/inputs/zipkin/handler_test.go:66]: Remove?

File: plugins/inputs/zipkin/handler_test.go [hunk 10]
```diff
@@ -81,7 +81,7 @@ func TestSpanHandler(t *testing.T) {
 			Timestamp:   time.Unix(0, 1498688360904552*int64(time.Microsecond)).UTC(),
 			Duration:    time.Duration(50410) * time.Microsecond,
 			ServiceName: "trivial",
-			Annotations: []trace.Annotation{},
+			Annotations: make([]trace.Annotation, 0),
```
Comment (srebhan) [inline plugins/inputs/zipkin/handler_test.go:84]: Remove?

**Review by srebhan — COMMENTED** [review]
Thanks @zak-pawel! Only marked some spots where we might want to remove the initialization altogether...

File: plugins/inputs/sqlserver/sqlserver_test.go (hunk 3 above)
Comment (zak-pawel) [reply plugins/inputs/sqlserver/sqlserver_test.go:19]: Test would fail.

File: plugins/inputs/vsphere/vsan.go (hunk 8 above)
Comment (zak-pawel) [reply plugins/inputs/vsphere/vsan.go:391]: I'm not sure what would happen later with this request.

File: plugins/inputs/zipkin/handler_test.go (hunk 9 above)
Comment (zak-pawel) [reply plugins/inputs/zipkin/handler_test.go:66]: Test would fail.

File: plugins/inputs/zipkin/handler_test.go (hunk 10 above)
Comment (zak-pawel) [reply plugins/inputs/zipkin/handler_test.go:84]: Test would fail.

File: plugins/inputs/smart/smart_test.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/inputs/smart/smart_test.go]: Changed.

File: plugins/inputs/snmp/snmp_test.go (hunk 2 above)
Comment (zak-pawel) [reply plugins/inputs/snmp/snmp_test.go]: Removed

File: plugins/inputs/stackdriver/stackdriver.go (hunk 4 above)
Comment (zak-pawel) [reply plugins/inputs/stackdriver/stackdriver.go]: Removed

File: plugins/inputs/supervisor/supervisor_test.go (hunk 5 above)
Comment (zak-pawel) [reply plugins/inputs/supervisor/supervisor_test.go]: Removed

File: plugins/inputs/supervisor/supervisor_test.go (hunk 6 above)
Comment (zak-pawel) [reply plugins/inputs/supervisor/supervisor_test.go]: Removed

File: plugins/inputs/varnish/varnish.go (hunk 7 above)
Comment (zak-pawel) [reply plugins/inputs/varnish/varnish.go]: Removed

File: plugins/inputs/vsphere/vsan.go (hunk 8 above)
Comment (srebhan) [reply plugins/inputs/vsphere/vsan.go:391]: So let's keep it as is.

**Pull request merged**